nomenclate/core/configurator.py

```diff
@@ -38,150 +25,214 @@ def __init__(self, path=None):
+            preceding_depth (int): always returns a dictionary encapsulating the data
+                                    that traces back up the path for x depth
+                                    -1 for the full traversal back up the path
+                                    None is default for no traversal
         """
-        if options:
-            return self.list_section_options(section)
-        return self.get_data(section, subsection, raw=raw)
+        query_path = query_path if isinstance(query_path, list) else [query_path]
+        self.validate_query_path(query_path)
+        config_entry = self.get_path_entry_from_config(query_path)
+        query_result = self.config_entry_handler.format_query_result(config_entry, query_path, return_type, preceding_depth)
+        return query_result
 
-    def query_valid_entry(self, section='', subsection=''):
-        """ Function to check if the config section/subsection data exists
-        """
-        data_query = None
-        if not subsection:
-            data_query = self.data.get(section, '')
-        else:
-            data_query = self.data.get(section, {}).get(subsection, '')
-        if not data_query:
-            raise IOError('Section: %s and Subsection: %s not found in current config file...' % (section, subsection))
-        return True
+    def get_path_entry_from_config(self, query_path):
+        if not query_path:
+            return list(self.data)
+        cur_data = self.data
+        for child in query_path:
+            cur_data = cur_data.get(child)
+        return cur_data
 
-    def get_section(self, section=None, raw=False):
-        """
-        Returns (dict): dictionary of the entire contents of a specific section
-        """
-        return self.get(section, raw=raw)
+    @classmethod
+    def validate_config_file(cls, config_file):
+        if not os.path.isfile(config_file):
+            raise IOError('File %s is not a valid yml, ini or cfg file or does not exist' % config_file)
 
-    def get_subsection(self, section=None, subsection=None, raw=False):
-        """ Return the given section/subsection pair as a list
-        Args:
-            section (str): section to query
-            subsection (str): subsection to query
-        Returns (dict): list of the subsection results
-        """
-        return self.get(section=section, subsection=subsection, raw=False)
+        elif os.stat(config_file).st_size > 0:
+            raise IOError('File %s is empty' % config_file)
 
-    def get_subsection_as_dict(self, section=None, subsection=None):
-        """ Return the given section/subsection pair as a dictionary
-        Args:
-            section (str): section to query
-            subsection (str): subsection to query
-        Returns (dict): dictionary of the subsection results
-        """
-        return {section: {subsection: self.get_subsection(section=section, subsection=subsection)}}
+        with open(config_file, 'r') as f:
```

Comment:
I think this should exist somewhere else, there's "validation" and then there's opening and retrieving data.  I think they should be separate methods.  But since this method isn't being used let's worry about that later.